pkg/ebpf/compiler/compiler.cpp

```diff
@@ -0,0 +1,273 @@
+#include "compiler.h"
+#include "files.h"
+
+#include <iostream>
+#include <mutex>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/Support/TargetSelect.h>
+#include <clang/CodeGen/CodeGenAction.h>
+#include <clang/Driver/Job.h>
+#include <clang/Driver/Tool.h>
+#include <clang/Frontend/CompilerInstance.h>
+#include <clang/Lex/PreprocessorOptions.h>
+
+std::once_flag ClangCompiler::llvmInitialized;
+
+enum Architecture { PPC, PPCLE, S390X, ARM64, X86 };
+
+namespace {
+    std::map<std::string, std::unique_ptr<llvm::MemoryBuffer>> remapped_files_;
+}
+
+ClangCompiler::ClangCompiler(const char *name) :
+    llvmContext(std::make_unique<llvm::LLVMContext>()),
+    diagOpts(new clang::DiagnosticOptions()),
+    errStream(errString),
+    textDiagnosticPrinter(new clang::TextDiagnosticPrinter(errStream, diagOpts.get())),
+    diagnosticsEngine(new clang::DiagnosticsEngine(diagID, diagOpts, textDiagnosticPrinter.get(), false)),
+    defaultCflags({
```

Comment:
should we include `KBUILD_MODNAME` here instead of re-defining it in each header

Reply:
It is defined in `compiler.go`. The other places it is defined, are the BCC compiled probes.